src/main/java/org/sagebionetworks/bridge/exporter/dynamo/DynamoHelper.java

```diff
@@ -0,0 +1,133 @@
+    @Resource(name = "ddbParticipantOptionsTable")
+    public final void setDdbParticipantOptionsTable(Table ddbParticipantOptionsTable) {
+        this.ddbParticipantOptionsTable = ddbParticipantOptionsTable;
+    }
+
+    /** Schema table, used to get schemas to create Synapse tables and serialize health data. */
+    @Resource(name = "ddbSchemaTable")
+    public final void setDdbSchemaTable(Table ddbSchemaTable) {
+        this.ddbSchemaTable = ddbSchemaTable;
+    }
+
+    /** Study table, used to get study config, like linked Synapse project. */
+    @Resource(name = "ddbStudyTable")
+    public final void setDdbStudyTable(Table ddbStudyTable) {
+        this.ddbStudyTable = ddbStudyTable;
+    }
+
+    /**
+     * Returns the schema for the given key.
+     *
+     * @param metrics
+     *         metrics object, used to keep a record of "schemas not found"
+     * @param schemaKey
+     *         key for the schema to get
+     * @return the schema
+     * @throws IOException
+     *         if we fail to deserialize the schema
+     * @throws SchemaNotFoundException
+     *         if the schema doesn't exist
+     */
+    public UploadSchema getSchema(Metrics metrics, UploadSchemaKey schemaKey) throws IOException,
+            SchemaNotFoundException {
+        Item schemaItem = getSchemaCached(schemaKey);
+        if (schemaItem == null) {
+            metrics.addKeyValuePair("schemasNotFound", schemaKey.toString());
+            throw new SchemaNotFoundException("Schema not found: " + schemaKey.toString());
+        }
+        return UploadSchema.fromDdbItem(schemaItem);
+    }
+
+    // Helper method that encapsulates just the DDB call, cached with annotation.
+    @Cacheable(lifetime = 5, unit = TimeUnit.MINUTES)
+    private Item getSchemaCached(UploadSchemaKey schemaKey) {
+        return ddbSchemaTable.getItem("key", schemaKey.getStudyId() + ":" + schemaKey.getSchemaId(),
+                "revision", schemaKey.getRevision());
+    }
+
+    /**
+     * Gets the sharing scope for the given user.
+     *
+     * @param healthCode
+     *         health code of user to get sharing scope for
+     * @return the user's sharing scope
+     */
+    @Cacheable(lifetime = 5, unit = TimeUnit.MINUTES)
+    public SharingScope getSharingScopeForUser(String healthCode) {
+        // default sharing scope is no sharing
+        SharingScope sharingScope = SharingScope.NO_SHARING;
+
+        try {
+            Item participantOptionsItem = ddbParticipantOptionsTable.getItem("healthDataCode", healthCode);
+            if (participantOptionsItem != null) {
+                String participantOptionsData = participantOptionsItem.getString("data");
+                Map<String, Object> participantOptionsDataMap = DefaultObjectMapper.INSTANCE.readValue(
+                        participantOptionsData, DefaultObjectMapper.TYPE_REF_RAW_MAP);
+                String sharingScopeStr = String.valueOf(participantOptionsDataMap.get("SHARING_SCOPE"));
```

Comment:
Possibly break out the JSON parsing code from the retrieval of sharing scope since soon you'll be adding data groups, and external identifier needs to be added as well, all from this map